### DISCUSSION TITLE
Solved: How to copy a node and to paste it as a new node with same text and a link to #ID_<id of copied node>?

### DISCUSSION THREAD

**Discussion by echo21** [freeplane/freeplane#378] [Help] [answered]
Hi my co-mind mappers,

maybe I am to blind to find such a feature in freeplane 1.9.12:

Assume we have a mind map (see attached zip) with just 2 sibling nodes "a" (ID_0000000001) and "b" (ID_0000000002) under the root node. I want to select and copy node "b". Then I want to paste the copied node as a new child node (assume ID_0000000003) under node "a". The node ID_0000000003 shall have the text "b" and the link target "#ID_0000000002" (note the added **#** at the beginning; this makes it link to a node in the same map).

Can you tell me if and how this works out of the box?

I've just found ctrl+shift+v to open the "Paste as ..." dialog. There I just find 4 radio buttons. None of them allows the paste feature I am looking for.

In case this is not a feature of Freeplane 1.9.12:

1. FP team members, would you please take this post as a feature request?
2. I have some experience in writing groovy scripts. (How) do groovy scripts have access to the clipboard and the properties of the copied node? Knowing the answer of how I would be easily able to implement this feature myself.

Thanks in advance

`<map version="freeplane 1.9.8">`
`<!--To view this file, download free mind mapping software Freeplane from https://www.freeplane.org -->`
`  <node TEXT="root" FOLDED="false" ID="ID_0000000000" CREATED="1610381621824" MODIFIED="1652861329464">`
`    <node TEXT="a" FOLDED="false" ID="ID_0000000001" CREATED="1652860407944" MODIFIED="1652860409097">`
`      <node TEXT="b" ID="ID_0000000003" CREATED="1652860424525" MODIFIED="1652860434617" LINK="#ID_0000000002"/>`
`    </node>`
`    <node TEXT="b" ID="ID_0000000002" CREATED="1652860409529" MODIFIED="1652860410764"/>`
`  </node>`
`</map>`

[CopyAndPasteAsInternalLink.zip](https://github.com/freeplane/freeplane/files/8714429/CopyAndPasteAsInternalLink.zip)

**Comment by macmarrum** [comment, marked as answer]
Hi echo21

Thanks for the clear problem statement and the sample mind map.

It so happens that I had a similar need and resolved it by creating a script. Here's a version adapted to your needs (also attached below) https://github.com/macmarrum/freeplane-scripting/blob/main/src/scripts/pasteWithBacklink.groovy

[pasteWithBacklink.zip](https://github.com/freeplane/freeplane/files/8715104/pasteWithBacklink.zip)

It should cover most of use cases, although has some limitations:
- copied and target nodes must be in the same mind map
- among the copied nodes, only the topmost ones are pasted (no descendants)
- node conditional styles are ignored (aren't copied)

**Reply by echo21** [reply]
Hi macmarrum.

thank you very much! Your script does exactly what I was looking for.
